src/include/sof/ipc/common.h

```diff
@@ -53,8 +53,11 @@ struct ipc {
 	void *comp_data;
 
 	/* PM */
-	int pm_prepare_D3;	/* do we need to prepare for D3 */
-
+	int pm_prepare_D3	/* do we need to prepare for D3 */
+#if defined(__ZEPHYR__) && defined(CONFIG_PM_POLICY_CUSTOM)
+	__attribute__ ((deprecated))
+#endif
+	;
```

Comment:
Sorry, I don't get the link to Zephyr in this patch. It seems "task_mask.POWERDOWN" expresses the same information as "pm_prepare_D3", and is stored in the same struct. 

First, why is this change needed, can't IPC4 handle set pm_prepare_D3?

Second, if for style/naming reasons we want to change, I don't see any reason to keep pm_prepare_D3 around. Why not just remove and replace all instances of it with task_mask. It doesn't seem to be related to Zephyr at all.


PS Plus the commit is wrong, this is not specific to legacy cAVS platforms, but all SOF platforms (except Intel platforms using IPC4).

Reply:
@kv2019i I think the reason for the `IPC_TASK_POWERDOWN` flag is to make it a part of the https://github.com/thesofproject/sof/blob/73f6171bb3c8c5b1d0963883014508022445b997/src/ipc/ipc-common.c#L278 check